tests/integration/integration_suite_test.go

```diff
@@ -1,13 +1,316 @@
+func TestReaperDeploymentScenario(t *testing.T) {
+	// only runs when the -short flag is used
+	if !testing.Short() {
+		t.Skip("Skipped in non-short mode")
+	}
+	namespace := initializeCluster(t)
+	deployClusterForReaper(t, namespace)
+	testReaper(t, namespace)
+	cleanupCluster(t, namespace)
+}
+
+func testReaper(t *testing.T, namespace string) {
+	log.Println(Step("Testing Reaper..."))
+	repairId := triggerRepair(t, namespace)
+	waitForSegmentDoneAndCancel(t, repairId)
+}
+
+func deployClusterForReaper(t *testing.T, namespace string) {
+	log.Println(Info("Deploying K8ssandra and waiting for Reaper to be ready"))
+	DeployClusterWithValues(t, namespace, "default", "three_nodes_cluster_with_reaper.yaml")
+	checkResourcePresenceForReaper(t, namespace)
+	waitForReaperPod(t, namespace)
+	checkReaperRegistered(t, namespace)
+}
+
+func checkResourcePresenceForReaper(t *testing.T, namespace string) {
+	CheckResourceWithLabelIsPresent(t, namespace, "service", "app.kubernetes.io/managed-by=reaper-operator")
+	CheckClusterExpectedResources(t, namespace)
+}
+
+func waitForReaperPod(t *testing.T, namespace string) {
+	WaitForReaperPod(t, namespace)
+}
+
+func checkReaperRegistered(t *testing.T, namespace string) {
+	CheckKeyspaceExists(t, namespace, "reaper_db")
+	CheckClusterIsRegisteredInReaper(t, "k8ssandra")
+}
+
+func triggerRepair(t *testing.T, namespace string) string {
+	log.Println(Info("Starting a repair"))
+	return TriggerRepair(t, namespace, "reaper_db")
+}
+
+func waitForSegmentDoneAndCancel(t *testing.T, repairId string) {
+	log.Println(Info("Waiting for one segment to be repaired and canceling run"))
+	WaitForOneSegmentToBeDone(t, repairId)
+	CancelRepair(t, repairId)
+}
+
+// Medusa scenario (invoked with a specific backend name):
+// - Register a cluster with 1 node
+// - Potentially install backend specific dependencies (such as Minio)
+// - Create the backend credentials secret
+// - Create a keyspace and a table
+// - Load 10 rows and check that we can read 10 rows
+// - Perform a backup with Medusa
+// - Load 10 rows and check that we can read 20 rows
+// - Restore the backup
+// - Verify that we can read 10 rows
+// - Terminate the namespace and delete the cluster
+func TestMedusaDeploymentScenario(t *testing.T) {
+	// only runs when the -short flag is used
+	if !testing.Short() {
+		t.Skip("Skipped in non-short mode")
+	}
+	const backupName = "backup1"
+	backends := []string{"Minio", "S3"}
+	for _, backend := range backends {
+		t.Run(fmt.Sprintf("Medusa on %s", backend), func(t *testing.T) {
+			namespace := initializeCluster(t)
+			createMedusaSecretAndInstallDeps(t, namespace, backend)
+			deployClusterForMedusa(t, namespace, backend)
+			medusaSuccess := t.Run("Test backup and restore", func(t *testing.T) {
+				testMedusa(t, namespace, backend, backupName)
+			})
+			if medusaSuccess {
+				cleanupCluster(t, namespace)
+			}
+		})
+	}
+}
+
+func testMedusa(t *testing.T, namespace, backend, backupName string) {
```

Comment:
Well, we'll have leftover files from each test. But they'll be in different folders since I've enabled multi-tenancy and a new namespace name is generated for each run.
We just need to think of doing some cleanup from time to time (or create a GH action that will handle it for us, which would be even better).

Reply:
So if I run these minio tests against my local installation, I'll have garbage sitting in my minio installation? Who is going to clean that?

Reply:
Let's take discussion of cleaning up backup storage outside of this PR. It is already. It is already hard enough to follow the all the discussions here, and we can address the cleanup separately in future work.